src/pages/account/center/index.tsx

```diff
@@ -0,0 +1,247 @@
+import Projects from './components/Projects';
+import Articles from './components/Articles';
+import Applications from './components/Applications';
+import { CurrentUser, TagType } from './data.d';
+import styles from './Center.less';
+
+const operationTabList = [
+  {
+    key: 'articles',
+    tab: (
+      <span>
+        文章 <span style={{ fontSize: 14 }}>(8)</span>
+      </span>
+    ),
+  },
+  {
+    key: 'applications',
+    tab: (
+      <span>
+        应用 <span style={{ fontSize: 14 }}>(8)</span>
+      </span>
+    ),
+  },
+  {
+    key: 'projects',
+    tab: (
+      <span>
+        项目 <span style={{ fontSize: 14 }}>(8)</span>
+      </span>
+    ),
+  },
+];
+
+interface CenterProps extends RouteChildrenProps {
+  dispatch: Dispatch<any>;
+  currentUser: CurrentUser;
+  currentUserLoading: boolean;
+}
+interface CenterState {
+  newTags: TagType[];
+  tabKey: 'articles' | 'applications' | 'projects';
+  inputVisible: boolean;
+  inputValue: string;
+}
+
+@connect(
+  ({
+    loading,
+    accountAndcenter,
+  }: {
+    loading: { effects: { [key: string]: boolean } };
+    accountAndcenter: ModalState;
+  }) => ({
+    currentUser: accountAndcenter.currentUser,
+    currentUserLoading: loading.effects['accountAndcenter/fetchCurrent'],
+  }),
+)
+class Center extends PureComponent<CenterProps, CenterState> {
+  // static getDerivedStateFromProps(
+  //   props: accountAndcenterProps,
+  //   state: accountAndcenterState,
+  // ) {
+  //   const { match, location } = props;
+  //   const { tabKey } = state;
+  //   const path = match && match.path;
+
+  //   const urlTabKey = location.pathname.replace(`${path}/`, '');
+  //   if (urlTabKey && urlTabKey !== '/' && tabKey !== urlTabKey) {
+  //     return {
+  //       tabKey: urlTabKey,
+  //     };
+  //   }
+
+  //   return null;
+  // }
+
+  state: CenterState = {
+    newTags: [],
+    inputVisible: false,
+    inputValue: '',
+    tabKey: 'articles',
+  };
+
+  public input: Input | null | undefined = undefined;
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'accountAndcenter/fetchCurrent',
+    });
+    dispatch({
+      type: 'accountAndcenter/fetch',
+    });
+  }
+
+  onTabChange = (key: string) => {
+    // If you need to sync state to url
+    // const { match } = this.props;
+    // router.push(`${match.url}/${key}`);
+    this.setState({
+      tabKey: key as CenterState['tabKey'],
+    });
+  };
+
+  showInput = () => {
+    this.setState({ inputVisible: true }, () => this.input && this.input.focus());
+  };
+
+  saveInputRef = (input: Input | null) => {
+    this.input = input;
+  };
+
+  handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ inputValue: e.target.value });
+  };
+
+  handleInputConfirm = () => {
+    const { state } = this;
+    const { inputValue } = state;
+    let { newTags } = state;
+    if (inputValue && newTags.filter(tag => tag.label === inputValue).length === 0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['magic numbers' are not allowed: 0](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)